### DISCUSSION TITLE
Support for USB to paralell port converter connected printers?

### DISCUSSION THREAD

**Discussion by carlsa** [michaelrsweet/lprint#148] [Q&A] [answered]
I'm trying to connect an old Zebra Z4000 with LPrint but am having problems getting it to work.

The printer is connected with a `QinHeng Electronics CH340S` (1a86:7584) USB to parallel adapter, this works in CUPS where it is automatically recognized with the URI `usb://Zebra%20Technologies/ZTC%20Z4000-300dpi`, but it doesn't show up in LPrint if I do `lprint devices`.

It also doesn't work when I try to add it with `lprint add` 
```
lprint add -d zebra_test -v usb://Zebra%20Technologies/ZTC%20Z4000-300dpi -m zpl_4inch-203dpi-dt
lprint: Unable to add printer: Unsupported smi55357-device-uri uri value.
```

I'm runing it on a Rapberry pi (Raspbian) where I build `PAPPLE 1.4.6` and `LPrint 1.3.1` from sources. 

So my question is if this is expected to work and, if not, where I could add it so it is recognized in LPrint?

**Comment by michaelrsweet** [comment]
There isn't anything special about this, but you generally need to run lprint as root ("sudo lprint devices") for USB access on Linux.

**Reply by carlsa** [reply, marked as answer]
I solved it, apparently it wasn't being built with support for USB devices, `configure --enable-libusb` before building helped.

For someone else that is troubleshooting something similar, this is how I found the problem:
- Printers seem to be detected by PAPPL and the USB devices are filtered for printers in `libusb_get_device_list`.
- I could build PAPPL with `configure --enable-debug-printfs` and saw that I didn't get the debug messages that `libusb_get_device_list` would generate and this suggested that it wasn't active.
- `configure --help` showed a list with options I could enable and `--enable-libusb` worked

Before doing that, I verified that the USB device didn't use the `usblp` subsystem with `udevadm info --query=all --name=/dev/usb/lp0`

I also verified that the conditions that something is detected as a printer from `libusb_get_device_list` was meet by looking at the USB device information with `lsusb -v`.
